pkgs/tools/filesystems/jfsutils/default.nix

```diff
@@ -24,6 +24,14 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoreconfHook ];
   buildInputs = [ libuuid ];
 
+  # Replace hard links with symlinks for reproducibility
```

Comment:
I wonder if this should be a stdenv default?

I once upon a time did a very similar thing to resolve absolute store path symlinks with relative ones (to reduce the differences across rebuilds that produce the exact same output but link somewhere).

Reply:
I did consider this (https://github.com/NixOS/nixpkgs/issues/114331#issuecomment-786016848), but I wasn't sure there is a one-size-fits-all solution: a stdenv hook would have to decide whether to duplicate or symlink the files, and in the case of symlinks decide which is the "main" file that the others should link to.

Because of this I figured it might be better to fix this on a case-by-case basis. What do you think the default behavior should be?

Reply:
hm, perhaps fail packages if they have hardlinks then? That would be better than silent failure (or reproducibility).